e2e/content-services/share-file/share-file.e2e.ts

```diff
@@ -199,5 +201,19 @@ describe('Share file', () => {
             await BrowserActions.getUrl(sharedLink);
             await viewerPage.checkFileNameIsDisplayed(pngFileModel.name);
         });
+
+        it('[C260153] Should shared files listed in share files custom resources', async () => {
+            await contentListPage.selectRow(pngFileModel.name);
+            await contentServicesPage.clickShareButton();
+            await shareDialog.checkDialogIsDisplayed();
+            await shareDialog.clickShareLinkButton();
+
+            await BrowserActions.closeMenuAndDialogs();
+            await browser.sleep(30000); //it get really long to update the shared link file list
```

Comment:
why not 27000 ? :)

Reply:
there must be another way of waiting, like maybe querying the server a few times until total number of items is the expected one

Reply:
I don't see how this thing could improve the test

Reply:
having hard coded values for browser.sleep is never a good idea.

This test will always wait 30 secs, no matter what.

I did a check using a retry mechanism that queries the backend every 1 sec, and the average I got was around 5-6 seconds. 
Of course, at times it might take longer, but at times it can also take less. Smart waits are always prefered instead of waiting for a fixed amount of time, and 30 seconds for a single test is a lot!

Reply:
@adinapitul do you already have this mechanism ?

Reply:
yes, see this in ACA:
https://github.com/Alfresco/alfresco-content-app/blob/development/e2e/utilities/repo-client/apis/shared-links/shared-links-api.ts#L92